### PR TITLE
Enable more linting rules

### PR DESCRIPTION
**Pull request by gabegorelick**
Hopefully these should be fairly uncontroversial.

Follow on from #477.

### CODE REVIEW

File: papaparse.js [hunk 1]
```diff
@@ -615,11 +615,11 @@
 				this._chunkError();
 			else
 				this._start += this._config.chunkSize;
-		}
+		};
 
 		this._chunkLoaded = function()
 		{
-			if (xhr.readyState != 4)
+			if (xhr.readyState !== 4)
```
Comment (gabegorelick) [inline papaparse.js]: If `readyState` were the string `"4"`, then this change would break. But I don't think that's possible.

File: papaparse.js [hunk 2]
```diff
@@ -1173,7 +1174,7 @@
 			comments = false;
 
 		// Newline must be valid: \r, \n, or \r\n
-		if (newline != '\n' && newline != '\r' && newline != '\r\n')
+		if (newline !== '\n' && newline !== '\r' && newline !== '\r\n')
```
Comment (gabegorelick) [inline papaparse.js]: Let me know if these were intentional for performance.

File: tests/test-cases.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
+var chai;
+var Papa;
```
Comment (gabegorelick) [inline tests/test-cases.js]: We should think about switching to let/const in the future.

**Comment by pokoli** [comment]
Did you mind adding some notes about the rules that we are checking on the how to contribute section of the README?

**Comment by gabegorelick** [comment]
> Did you mind adding some notes about the rules that we are checking on the how to contribute section of the README?

I'd be happy to, but not sure exactly what I can add above the docs in https://eslint.org/docs/rules/.

**Pull request merged**